### PR TITLE
#292 Create reusable API reducer

### PR DESCRIPTION
**Pull request by josephj**
#### Description

Add a root reducer which handles the API actions

#### QA URL

https://demo-roster.efcsydney.org

#### Acceptance Criteria

- [ ] Ensure the usage is straight-forward and easy to understand
- [ ] Ensure we have new root props in redux store (`@resource`, `@cache`, and `@request`)
- [ ] Ensure the everything is still working good

### CODE REVIEW

File: client/src/resource/reducer.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+            item.account = payload.params.account;
+          }
+        });
+      }
+      const idAttribute = apiMapping[resource.name].idAttribute;
+
+      // not idAttribute means it's a single resource
+      if (idAttribute) {
+        let normalizeSchema = new Schema(resource.name, {
+          idAttribute: apiMapping[resource.name].idAttribute
```
Comment (trust2065) [inline client/src/resource/reducer.js]: We can use `idAttribute` define above to replace `apiMapping[resource.name].idAttribute` ?

Comment (josephj) [reply client/src/resource/reducer.js]: Sorry, I don't understand what do you mean. Can you explain a bit more? 😬

The idAttribute definition is here. 
https://github.com/efcsydney/efcsydney-roster/pull/293/files#diff-73f41bf011fd3bc92b37a72b8034e755R31

Comment (trust2065) [reply client/src/resource/reducer.js]: I found `idAttribute` has been assigned as `apiMapping[resource.name].idAttribute;` in line 132.
That why I think we can replace line 137
`idAttribute: apiMapping[resource.name].idAttribute` by 
`idAttribute: idAttribute`

Comment (josephj) [reply client/src/resource/reducer.js]: I will do it in another pull request.

**Pull request merged**